### ISSUE TITLE
Failed to import necessary dependency botocore, please install botocore first

### ISSUE THREAD

**Issue by AhmadMasry** [kubernetes-sigs/aws-efs-csi-driver#950]
/kind bug

When using 1.5.2 the mount of efs volumes on pods fails with the following error:
MountVolume.SetUp failed for volume "XXXX" : rpc error: code = Internal desc = Could not mount "fs-0d5b5e6122e910dce:/" at "/var/lib/kubelet/pods/XXXX/volumes/kubernetes.io~csi/pvc-XXXX/mount": mount failed: exit status 1 Mounting command: mount Mounting arguments: -t efs -o accesspoint=fsap-XXXX,tls fs-XXXX:/ /var/lib/kubelet/pods/XXXX/volumes/kubernetes.io~csi/pvc-XXXX/mount Output: Failed to resolve "fs-XXXX.efs.XXXX.amazonaws.com" - check that your file system ID is correct, and ensure that the VPC has an EFS mount target for this file system ID. See https://docs.aws.amazon.com/console/efs/mount-dns-name for more detail. Attempting to lookup mount target ip address using botocore. Failed to import necessary dependency botocore, please install botocore first.

And on the node daemonset pod the following error:
E0228 19:50:42.116875       1 driver.go:103] GRPC error: rpc error: code = Internal desc = Could not mount "fs-XXXX:/" at "/var/lib/kubelet/pods/XXXX/volumes/kubernetes.io~csi/pvc-XXXX/mount": mount failed: exit status 1
Mounting command: mount
Mounting arguments: -t efs -o accesspoint=fsap-XXXX,tls fs-XXXX:/ /var/lib/kubelet/pods/XXXX/volumes/kubernetes.io~csi/pvc-XXXX/mount
Output: Failed to resolve "fs-XXXX.efs.us-west-2.amazonaws.com" - check that your file system ID is correct, and ensure that the VPC has an EFS mount target for this file system ID.
See https://docs.aws.amazon.com/console/efs/mount-dns-name for more detail.
Attempting to lookup mount target ip address using botocore. Failed to import necessary dependency botocore, please install botocore first.

The issue is solved for me by rolling back to 1.5.1

- Kubernetes version (use `kubectl version`):
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.8-eks-ffeb93d", GitCommit:"abb98ec0631dfe573ec5eae40dc48fd8f2017424", GitTreeState:"clean", BuildDate:"2022-11-29T18:45:03Z", GoVersion:"go1.18.8", Compiler:"gc", Platform:"linux/amd64"}
- Driver version: 1.5.2

**Comment by jfmeachum** [comment]
I'm seeing this in driver versions 1.4.9, 1.5.1 and 1.5.2

```
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.8-eks-ffeb93d", GitCommit:"abb98ec0631dfe573ec5eae40dc48fd8f2017424", GitTreeState:"clean", BuildDate:"2022-11-29T18:45:03Z", GoVersion:"go1.18.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by jfmeachum** [comment]
I had to roll back to [helm chart version 2.3.7](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/helm-chart-aws-efs-csi-driver-2.3.7/charts/aws-efs-csi-driver/Chart.yaml), application version 1.5.0.

The [Dockerfile of 1.5.1](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/helm-chart-aws-efs-csi-driver-2.3.8/Dockerfile#L91) looks like it changed the way botocore was being installed via pip. It copies from a prior stage to `/usr/local/lib/python3.9/site-packages/`, however that path is not on the `sys.path` because it looks like the `python3` on the final stage is using python3.7 and not the expected python3.9.

```console
$ docker run --entrypoint /bin/bash -it amazon/aws-efs-csi-driver:v1.5.1

bash-4.2# python
Python 2.7.18 (default, May 25 2022, 14:30:51) 
[GCC 7.3.1 20180712 (Red Hat 7.3.1-15)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys; sys.path
['', '/usr/lib/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib/python2.7/site-packages']

bash-4.2# python3
Python 3.7.16 (default, Dec 15 2022, 23:24:54) 
[GCC 7.3.1 20180712 (Red Hat 7.3.1-15)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys; sys.path
['', '/usr/lib64/python37.zip', '/usr/lib64/python3.7', '/usr/lib64/python3.7/lib-dynload', '/usr/local/lib/python3.7/site-packages', '/usr/lib64/python3.7/site-packages', '/usr/lib/python3.7/site-packages']

```

**Comment by max-blue** [comment]
The issue persist in v1.5.3 - I just installed it on my K8S cluster and running into the same botocore missing issue along with some additional errors.

**Comment by RyanStan** [comment]
@max-blue How are you installing the driver?  Can you describe the Kubernetes objects that are deployed to confirm the image version that has been deployed?  E.g. `kubectl describe daemonset.apps/efs-csi-node -n kube-system`.

We've confirmed that [the latest v1.5.3 image on the public ECR](https://gallery.ecr.aws/efs-csi-driver/amazon/aws-efs-csi-driver) has botocore installed in the python path.

**Comment by AhmadMasry** [comment]
I actually can confirm that the latest helm chart with the 1.5.3 has worked fine for me.

**Comment by max-blue** [comment]
> @max-blue How are you installing the driver? Can you describe the Kubernetes objects that are deployed to confirm the image version that has been deployed? E.g. `kubectl describe daemonset.apps/efs-csi-node -n kube-system`.
> 
> We've confirmed that [the latest v1.5.3 image on the public ECR](https://gallery.ecr.aws/efs-csi-driver/amazon/aws-efs-csi-driver) has botocore installed in the python path and have confirmed that mounts are working when dns resolution is disabled in a vpc (to force botocore to be used).
> 
> Additionally, you mentioned other errors. Feel free to include those here as well, or open a separate issue with them.

It turns out the helm chart version 1.5.3 uses the 1.5.2 image tag hence why I run into the issue.

**Comment by mskanth972** [comment]
Closing as the issue was resolved in the v1.5.3
